app/controllers/profiles_controller.rb

```diff
@@ -2,16 +2,27 @@ class ProfilesController < ApplicationController
   def show
     @profile = Profile.find(params[:id])
   end
-
-  def search_gender(term)
-    Profile.where(gender: term)
-  end
   
   def search
     
   end
   
   def search_results
-    @profile_results = search_gender(params[:gender])
+    @profile_results = search_profiles(params[:search])
+  end
+
+  def search_profiles(params)
```

Comment:
If you don't intend search_profiles to be a controller action, I suggest making this method private.